corehq/apps/users/models.py

```diff
@@ -2615,7 +2611,8 @@ def send_request_email(self):
                                     email_from=settings.DEFAULT_FROM_EMAIL)
 
 
-class SQLInvitation(SyncSQLToCouchMixin, models.Model):
+class Invitation(models.Model):
+    uuid = models.UUIDField(primary_key=True, db_index=True, default=uuid4)
```

Comment:
Would it be difficult to rename this to `id`?

Reply:
I'd prefer not to because `id` in our code is typically an integer.

Reply:
Ah, I just saw https://github.com/dimagi/commcare-hq/pull/26975#issuecomment-605223930. I don't agree with the reasoning, I respect your decision.

For my $0.02, I think there's a stronger convention that the primary key is named `id` than that that column always has the same data type (auto-incrementing integer).

Reply:
Cool, I see your reasoning but am going to keep it as is. I need an "agree to disagree" emoji. 👩‍⚖ ?

Reply:
FWIW Django _does_ alias `obj.pk` to whatever the primary key field is, whether `id`, `uuid`, or otherwise.

Reply:
@esoergel true, but I've run into issues with `pk` in the past. It does not work as expected in every scenario (although the ones where it doesn't are slipping my mind right now).

Reply:
I'm curious about issues with `pk` because I use it too. I like it because it standardizes the public API of models while allowing them to name their primary key after its data type or origin. It also makes my intention clearer, e.g. "I want this Location. The nature of the identifier is irrelevant." vs. "I want the Location whose Couch ID was "f00"."

Just to summarize where we are at:

attribute | HQ convention
--- | ---
id | An auto-incrementing integer
uuid | A UUID
foo_id | ID based on original Couch model for a class Foo
pk | An alias of the primary key field

(fwiw: One day Django models [might](https://code.djangoproject.com/ticket/373) support [multi-column primary keys](https://code.djangoproject.com/wiki/MultipleColumnPrimaryKeys) (SQLAlchemy [already does](https://docs.sqlalchemy.org/en/13/core/constraints.html#primary-key-constraint) ) in which case `pk` would likely be a tuple for models with multi-column primary keys.)

Reply:
@millerdev are you thinking of related models, perhaps?  For instance:
```python
>>> location.location_type_id == location.location_type.id
True
>>> location.location_type_pk == location.location_type.pk
AttributeError: 'SQLLocation' object has no attribute 'location_type_pk'
```

Reply:
@esoergel that doesn't ring a bell.

@kaapstorm my thoughts here did not relate to Couch ids, but my preference there is to name such a column `couch_id` rather than `foo_id` (where `Foo` is the model name) since that makes it more obvious where it came from. We have code that follows this convention, although I'm not sure if we have always been consistent.

I'll try to keep this discussion in mind and let you know next time I come across a place where `model.pk` doesn't work as expected. One thing that does come to mind is raw SQL statements, where Django is not your code and the database (obviously you can't use `pk` there). I still think it's a good convention to use  `id` as the primary key name for SQL models having a single-column key, regardless of the column data type.

Reply:
I also prefer `couch_id` instead of `foo_id` because it's explicit. Messaging uses `couch_id`, but most models use the `foo_id` convention.

Reply:
(Oh, just to be clear, I would rather follow convention than my personal preference. I have worked on a team that applied their personal preferences without consensus. We had a Perl library that used four different indentation styles (2-space, 3-space, 4-space and tabs), and several different ways of passing function parameters, **in the same file**.)

Reply:
That's heinous. And I *like* perl.